### DISCUSSION TITLE
I need the fetch api in pocketbase to send the same header as the express.

### DISCUSSION THREAD

**Discussion by sairash** [pocketbase/pocketbase#624] [Q&A]
I am making an express api using pocketbase. 
How do I send the same headers as the express is getting?

Request and Response
![image](https://user-images.githubusercontent.com/29134272/192059339-f6b3e004-c4bc-4757-842b-6df05865032d.png)

Gets routed
![image](https://user-images.githubusercontent.com/29134272/192059519-88e10a88-8834-44fe-92ec-37761af44525.png)

And finally sends and gets data from pocketbase
![image](https://user-images.githubusercontent.com/29134272/192059778-bbba0f2f-9ad4-486c-9d4c-3c971537ae0a.png)

I need the fetch api in pocketbase to send the same header as the express.

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the question.

If you are asking for how to authenticate:

```js
// authenticate as regular user
await client.users.authViaEmail("test_user@example.com", "123456");

// authenticate as super-admin
await client.admins.authViaEmail("test_admin@example.com", "123456");

// fetch your user
await client.users.getOne("USER_ID");
```

The `Authorization` header will be auto set for you once you authenticate as user or admin. You could also check the Authentication docs in https://pocketbase.io/docs/api-authentication/.

Note that when authenticated as regular user you are restricted to only the records and actions that the user is allowed to.

Super-admins on the other hand have access to everything and can perform any action (including deleting or creating other admins).

**Reply by ganigeorgiev** [reply]
If you are asking for how to pass custom headers to the `client.users.getOne("USER_ID")` call - currently you can't. You can use the bare send call if you need to make custom request:

```js
await client.send("/api/users/USER_ID", {
    // for list of all supported options - https://developer.mozilla.org/en-US/docs/Web/API/fetch#options
    "headers": {
        "User-Agent": "..."
    }
});
```

**Reply by sairash** [reply]
I guess this is the best way right now.. But is there a query builder for what links to send in client.send ??

`await client.send( someQueryBuilderFunction() , {
    // for list of all supported options - https://developer.mozilla.org/en-US/docs/Web/API/fetch#options
    "headers": {
        "User-Agent": "..."
    }
});`

**Reply by ganigeorgiev** [reply]
> But is there a query builder for what links to send in client.send 

No, because usually this shouldn't be needed. Based on your examples, you seems to want to create a wrapper API, which is a valid use case, but it is not how the SDK was intended to be used. _JS supports "external patching", so you can always attach/overwrite existing methods via the `PocketBase.prototype.*`_.

**Comment by sairash** [comment]
Isn't it possible to authenticate only using tokens?

**Reply by ganigeorgiev** [reply]
If you mean via API keys - No. I may consider it in the future since this was asked previously, but for now if you need to perform some additional server-side only system operation, you can authenticate as admin.
